src/Mutator/Number/DecrementInteger.php

```diff
@@ -98,6 +98,10 @@ private function isAllowedComparison(Node\Scalar\LNumber $node): bool
 
         $parentNode = ParentConnector::getParent($node);
 
+        if ($parentNode instanceof Node\Expr\ArrayDimFetch && $node->value === 0) {
```

Comment:
I don't think this should be inside `isAllowedComparison()` method, because this is not related to `comparison`

Could you please move it e.g. before line 86 (`if ($this->isPartOfSizeComparison($node)) {`)